### PR TITLE
MINIFICPP-1687 Signal error on UUID collision

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-1687

-------------------------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by arpadboda — APPROVED** [review]

File: libminifi/src/core/yaml/YamlConfiguration.cpp [hunk 1]
```diff
@@ -908,6 +914,14 @@ YAML::Node YamlConfiguration::getOptionalField(const YAML::Node& yamlNode, const
   return result;
 }
 
+void YamlConfiguration::addNewId(const std::string& uuid) {
+  if (uuids_.find(uuid) != uuids_.end()) {
+    throw Exception(ExceptionType::GENERAL_EXCEPTION, "UUID " + uuid + " is duplicated in the flow configuration");
+  }
+
+  uuids_.insert(uuid);
```
Comment (fgerlits) [inline libminifi/src/core/yaml/YamlConfiguration.cpp:922]: minor, but `insert()` returns whether the insertion was successful, so we could avoid doing two lookups:
```suggestion
  const auto [_, success] = uuids_.insert(uuid);
  if (!success) {
    throw Exception(ExceptionType::GENERAL_EXCEPTION, "UUID " + uuid + " is duplicated in the flow configuration");
  }
```

File: libminifi/include/core/yaml/YamlConfiguration.h [hunk 2]
```diff
@@ -277,6 +278,7 @@ class YamlConfiguration : public FlowConfiguration {
    * @return         the parsed or generated UUID string
    */
   std::string getOrGenerateId(const YAML::Node& yamlNode, const std::string& idField = "id");
+  std::string getRequiredIdField(const YAML::Node& yaml_node, std::string_view yaml_section = "", std::string error_message = "");
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply libminifi/include/core/yaml/YamlConfiguration.h]: Good point, it parameter list was copied from `checkRequiredField` and should have been changed. Fixed in 33c6c28c702f72e67add5c9dfd318020019a1788

File: libminifi/src/core/yaml/YamlConfiguration.cpp (hunk 1 above)
Comment (lordgamez) [reply libminifi/src/core/yaml/YamlConfiguration.cpp]: Good idea, updated in 33c6c28c702f72e67add5c9dfd318020019a1788

**Review by fgerlits — APPROVED** [review]

File: libminifi/include/core/yaml/YamlConfiguration.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/include/core/yaml/YamlConfiguration.h]: I would prefer `std::string_view` instead. It avoids copying and doesn't force the caller to allocate a `std::string`.

[the remark this answers is not in the document]
Comment (lordgamez) [reply libminifi/include/core/yaml/YamlConfiguration.h]: Replaced the error_message string to `std::string_view` int required field checks in 3bdcc32d44536916be6102c6e2b17dc67c7975bd

**Review by adamdebreceni — APPROVED** [review]

**Pull request closed without merging**